source/cgal/intersections.cc

```diff
@@ -316,26 +314,24 @@ namespace CGALWrappers
 
 
     std_cxx17::optional<std_cxx17::variant<CGALPoint3, CGALSegment3>>
-    compute_intersection(const std::array<Point<3>, 2> &first_simplex,
-                         const std::array<Point<3>, 4> &second_simplex)
+    compute_intersection_tetra_segment(
+      const std::array<Point<3>, 2> &tetrahedron,
+      const std::array<Point<3>, 4> &segment)
     {
 #  if DEAL_II_CGAL_VERSION_GTE(5, 5, 0)
       std::array<CGALPoint3, 4> pts0;
       std::array<CGALPoint3, 2> pts1;
       std::transform(
-        first_simplex.begin(),
-        first_simplex.end(),
+        tetrahedron.begin(),
+        tetrahedron.end(),
         pts0.begin(),
         [&](const Point<3> &p) {
           return CGALWrappers::dealii_point_to_cgal_point<CGALPoint3>(p);
         });
 
 
       std::transform(
-        second_simplex.begin(),
-        second_simplex.end(),
-        pts1.begin(),
-        [&](const Point<3> &p) {
+        segment.begin(), segment.end(), pts1.begin(), [&](const Point<3> &p) {
           return CGALWrappers::dealii_point_to_cgal_point<CGALPoint3>(p);
         });
```

Comment:
This will not compile because the template parameter `UnaryOperation` from [std::transform](https://en.cppreference.com/w/cpp/algorithm/transform) will not be deduced.

Reply:
No, there should be no problem. The `UnaryType` is simply the type of the pointer-to-function `&CGALWrappers::dealii_point_to_cgal_point<CGALPoint3>`. Or am I missing something?

Reply:
The return type of the function pointer is the function signature which includes ```Point<dim>``` (which the compiler can not deduce). Thus, we need ```&CGALWrappers::dealii_point_to_cgal_point<CGALPoint3,dim>```